### DISCUSSION TITLE
Guest behaviour while Queuing

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#18383] [Feature requests]
I think it'd be neat to see the following implemented, would add a bit more depth to gameplay:
If a peep is close to the entrance of the ride it's less likely for them to leave the queue.
If a ride is broken down it will be more likely for people to leave a queue, the banner should say "Broken Down" instead of "Closed" and peeps can still queue but it's way less likely.
During rain: Instead of nobody queueing, less people queue, but still queue (especially if they have an umbrella, or if the queue has a roof/is underground). If the queue piece isn't covered it's more likely for peeps to also leave the queue, and other peeps to will stand still (90% of the time) under a covered piece of the queue if they do not have an umbrella.
During a thunderstorm: Same peep functionality as broken down, except the ride still says open.

I don't think this change is too invasive to vanilla rct players, so I doubt being able to switch between them would be necessary.

**Comment by ocalhoun6** [comment]
>If a peep is close to the entrance of the ride it's less likely for them to leave the queue.

This makes a ton of sense and should probably be implemented. This way, peep behavior makes much more sense. Matches very well with real-world behavior.

>If a ride is broken down it will be more likely for people to leave a queue,

This also makes a lot of sense.

>the banner should say "Broken Down" instead of "Closed" and peeps can still queue but it's way less likely.

This less so. In a real park, the park probably wouldn't advertise that the ride is broken down that way, and they probably *would* just say it was "temporarily closed" with no further explanation to guests. Proclaiming publicly that it's broken down might give the impression that their maintenance is inadequate or even that the ride might be unsafe.

As for peeps joining the queue of a broken-down ride... To make any sense, that would have to be *very* rare indeed ... and at that point, is it really worth the coding work to implement it?

>During rain: Instead of nobody queueing, less people queue, but still queue (especially if they have an umbrella, or if the queue has a roof/is underground). If the queue piece isn't covered it's more likely for peeps to also leave the queue, and other peeps to will stand still (90% of the time) under a covered piece of the queue if they do not have an umbrella.
>During a thunderstorm: Same peep functionality as broken down, except the ride still says open.

These weather-based changes seem like too big of a change to me.
